zebra-consensus/src/config.rs

```diff
@@ -30,9 +35,37 @@ pub struct Config {
     /// For security reasons, this option might be deprecated or ignored in a future Zebra
     /// release.
     pub checkpoint_sync: bool,
+}
+
+impl From<InnerConfig> for Config {
+    fn from(
+        InnerConfig {
+            checkpoint_sync, ..
+        }: InnerConfig,
+    ) -> Self {
+        Self { checkpoint_sync }
+    }
+}
+
+impl From<Config> for InnerConfig {
+    fn from(Config { checkpoint_sync }: Config) -> Self {
+        Self {
+            checkpoint_sync,
+            ..Default::default()
+        }
+    }
+}
+
+/// Inner consensus configuration for backwards compatibility with fields that have been removed.  
+#[derive(Clone, Debug, Eq, PartialEq, Default, Deserialize, Serialize)]
+#[serde(deny_unknown_fields, default)]
```

Comment:
Moving the `default` attribute to a struct without a `Default` impl changes the default for `checkpoint_sync` from true to false.

Reply:
Whoops. Thank you. Fixed in https://github.com/ZcashFoundation/zebra/pull/7844/commits/8ef372dd22edddf66b9503e81b9e576b628ef657.